### ISSUE TITLE
TES unit tests

### ISSUE THREAD

**Issue by hannorein** [hannorein/rebound#675]
@PeterBartram: One TES unit test fails on my computer:
```
FAIL: test_integration_output_particles (rebound.tests.test_tes.TestIntegratorTES)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/rein/git/rebound/rebound/tests/test_tes.py", line 89, in test_integration_output_particles
    self.assertLess(de, 1e-15)
AssertionError: 1.039239216630799e-15 not less than 1e-15
```
However, it passes on the CI. That is a bit worrying because they results of the test should be the same, regardless where they are run. I don't think this is related, but FWIW, I just fixed a bug in the `rebound_integrator_tes` ctypes structure definition (what is actually a double was listed as a pointer to a double).

**Comment by PeterBartram** [comment]
Ah, that's not good, it sounds like undefined behaviour somewhere. I'll see if I can track down the problem. What OS are you seeing the failing test on?

**Comment by hannorein** [comment]
That's on MacOS.

**Comment by PeterBartram** [comment]
I got to the bottom of this one, I'd left a call to pow() in the code for the step size update. I have replaced it with ias15 version of  x^1/7. I have tested for bitwise energy violation matching for the longterm integration test is the unit test file (1e6 orbits for inner solar system) using a ubuntu desktop and an M1 macbook.

I have opened a pull request for the fix but wasn't sure if I was meant to link it to this issue somehow.
